### ISSUE TITLE
Over 80,000 Netlify function invocations

### ISSUE THREAD

**Issue by joepio** [ontola/legalsafetlylab#8]
> 83,930 out of 125,000 functions used. 41,070 requests left

> Your invocations usage on site legalsafetylab has reached 50% of the included allowance for Functions Level 0 (free) in the current billing cycle from October 18 to November 18.
> If it passes 100% before the end of the billing cycle, we’ll upgrade to [Functions Level 1](https://www.netlify.com/pricing/#functions). Check out your site dashboard for current usage details.

It's a ~~static~~ simple site! What's going on?

**Comment by Polleps** [comment]
It's not a static site... We discussed this already. You can't have a static site that also instantly reflects changes in the cms.
Still that's a lot of invocations. That definitely should not happen unless the site is really populair.

**Comment by joepio** [comment]
Yeah something odd is going on. I can't see the logs, but I'm assuming it's a lot of bot spam

**Comment by joepio** [comment]
Most days there is one moment / short period where a lot (up to 10k) requests are sent. 

<img width="1233" alt="Screenshot 2024-11-25 at 10 51 57" src="https://github.com/user-attachments/assets/1fcfd368-ba7f-4434-ae49-531afcd4f627">


IIRC most of these are things like `/wp-admin.php` and the likes. Some bot bombarding our server. Each request leads to a Query on `atomicdata.dev` I think.

**Comment by joepio** [comment]
I can't get any useful logs from Netlify that help me understand what is going on. I can't reproduce the `/wp-admin` stuff I mentioned above, I think because those were created by Sveltekit or something. Maybe Astro logging just sucks? 

<img width="1089" alt="Screenshot 2024-11-25 at 11 37 04" src="https://github.com/user-attachments/assets/15c7f3e0-6d36-417a-975f-c40a0a4930de">

**Comment by Polleps** [comment]
Caching should prevent this problem (unless each request is a different route) so maybe there is a problem with that. One thing I can think of is that maybe the experimental Astro environment API does not work on netlify. This API should check if an environment variable is set and if not set it with a default value. If this doesn't work the `CACHE_MAX_AGE` variable will never be set and the max age will be undefined. This could be fixed by manually setting the env variable in Netlify.

But since the logging is very bad it's difficult to say if this is actually what happens.

**Comment by joepio** [comment]
Just set this:

`CACHE_MAX_AGE=86400`

And redeployed
